solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java

```diff
@@ -818,6 +1193,12 @@ protected Integer parseStr(String rawval) {
     public Integer parseAndAddGap(@SuppressWarnings("rawtypes") Comparable value, String gap) {
       return ((Number) value).intValue() + Integer.parseInt(gap);
     }
+
+    @Override
+    @SuppressWarnings({"rawtypes"})
+    protected long toSortableDocValue(Comparable calcValue) {
+      return (long) calcValue;
+    }
```

Comment:
Your exception relates to this change here. Why did you change this to a direct cast?
The Solr 8 implementation had:
```
    @Override
    protected long toSortableDocValue(Comparable calcValue) {
      return ((Integer) calcValue).longValue();
    }
```

Reply:
Do you mind calling out any other changes you made for the patch?

Reply:
Solr 9.0 gradle has some strict checking on type conversion task. We have couple of other similar tickets for type conversion. I will look this further.